File: Euler_017/Euler_017.java
import java.math.*;
import java.util.*;

class Euler_017{
  static Map<Integer, Integer> wordCounts = new HashMap<Integer, Integer>(){{
    put(0,0);
    //one
    put(1,3);
    //two
    put(2,3);
    //three
    put(3,5);
    //four
    put(4,4);
    //five
    put(5,4);
    //six
    put(6,3);
    //seven
    put(7,5);
    //eight
    put(8,5);
    //nine
    put(9,4);
    //ten
    put(10,3);
    //eleven
    put(11,6);
    //twelve
    put(12,6);
    //thirteen
    put(13,8);
    //fourteen
    put(14,8);
    //fifteen
    put(15,7);
    //sixteen
    put(16,7);
    //seventeen
    put(17,9);
    //eighteen
    put(18,8);
    //nineteen
    put(19,8);
    //twenty
    put(20,6);
    //thirty
    put(30,6);
    //forty
    put(40,5);
    //fifty
    put(50,5);
    //sixty
    put(60,5);
    //seventy
    put(70,7);
    //eighty
    put(80,6);
    //ninety
    put(90,6);
    //One Hundred
    put(100,10);
    //Two Hundred
    put(200,10);
    //Three Hundred
    put(300,12);
    //Four Hundred
    put(400,11);
    //Five Hundred
    put(500,11);
    //Six Hundred
    put(600,10);
    //Seven Hundred
    put(700,12);
    //Eight Hundred
    put(800,12);
    //Nine Hundred
    put(900,11);
    //One Thousand
    put(1000,11);
  }};

  public static void main(String[] args) {
    long begin = System.currentTimeMillis();

    int number_of_letters = 0;

    for(int i = 1;i < 1001;i++){
        number_of_letters += processDigits(i);
      }
    //three hundred and three
    System.out.println(processDigits(303));
    System.out.println(number_of_letters);

    long end = System.currentTimeMillis();
		System.out.println(end-begin + "ms");
  }

  static int processDigits(int digits){
    int process_digit_letter_count = 0;
    String digitString = Integer.toString(digits);

    if (digits < 20){
      return wordCounts.get(digits);
    }

    for(int i = 0; i < digitString.length(); i++){
      char digit = digitString.charAt(i);
      int number = Character.getNumericValue(digit);

      if(digitString.length() == 4)
      {
        process_digit_letter_count += wordCounts.get(1000);
      }else if(digitString.length() > 2)
      {
        //hundreds number
        if(i == 0){
          number *= 100;
          process_digit_letter_count += wordCounts.get(number);
        }else if(i == 1){
          //tens number
          number *= 10;
          process_digit_letter_count += wordCounts.get(number);
          //add and
          process_digit_letter_count += (digits % 100 != 0 ? 3 : 0);
        }else if (i == 2){
          //ones number
          process_digit_letter_count += wordCounts.get(number);
        }
      }else
      {
        //tens number
        if(i == 0){
          number *= 10;
          process_digit_letter_count += wordCounts.get(number);
        }else{
          //ones number
          process_digit_letter_count += wordCounts.get(number);
        }
      }

    }
    return process_digit_letter_count;
  }
}

File: Euler_017/Euler_017.swift
//: my Dirty Code. Refactor later

import UIKit

var numbers = (one:"one",two:"two",three:"three", four:"four", five:"five", six:"six", seven:"seven", eight:"eight", nine:"nine", ten:"ten", eleven:"eleven", twelve:"twelve", thirteen:"thirteen", fourteen:"fourteen", fifteen:"fifteen", sixteen:"sixteen", seventeen:"seventeen", eighteen:"eighteen",nineteen:"nineteen", twenty:"twenty", thirty:"thirty", forty:"forty", fifty:"fifty", sixty:"sixty", seventy:"seventy", eighty:"eighty", ninety:"ninety", hundred:"hundred", thousand:"thousand")


var onetoNineteen = ["one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen"]
var tens = ["ten", "twenty", "thirty", "forty", "fifty", "sixty", "seventy", "eighty", "ninety"]

let hundred = "hundred"
let thousand = "thousand"


var count = 0

func countChar(temp : Int){
    switch temp {
    case 1:
        count = count + numbers.one.characters.count
        break
    case 2:
        count = count + numbers.two.characters.count
        break
    case 3:
        count = count + numbers.three.characters.count
    case 4:
        count = count + numbers.four.characters.count
    case 5:
        count = count + numbers.five.characters.count
    case 6:
        count = count + numbers.six.characters.count
    case 7:
        count = count + numbers.seven.characters.count
        break
    case 8:
        count = count + numbers.eight.characters.count
        break
    case 9:
        count = count + numbers.nine.characters.count
        break
    case 10:
        count = count + numbers.ten.characters.count
        break
    case 11:
        count = count + numbers.eleven.characters.count
        break
    case 12:
        count = count + numbers.twelve.characters.count
        break
    case 13:
        count = count + numbers.thirteen.characters.count
        break
    case 14:
        count = count + numbers.fourteen.characters.count
        break
    case 15:
        count = count + numbers.fifteen.characters.count
        break
    case 16:
        count = count + numbers.sixteen.characters.count
        break
    case 17:
        count = count + numbers.seventeen.characters.count
        break
    case 18:
        count = count + numbers.eighteen.characters.count
        break
    case 19:
        count = count + numbers.nineteen.characters.count
        break
    default:
        charCount(charCount: temp)
        break;
    }
}

func charCount(charCount: Int){
    var test = charCount
        if (test >= 20 && test < 30){
            count = count + numbers.twenty.characters.count
            test = test - 20;
            countChar(temp: test)
        }
        if (test >= 30 && test < 40){
            count = count + numbers.thirty.characters.count
            test = test - 30;
            countChar(temp: test)
        }
        if (test >= 40 && test < 50){
            count = count + numbers.forty.characters.count
            test = test - 40;
            countChar(temp: test)
        }
        if (test >= 50 && test < 60){
            count = count + numbers.fifty.characters.count
            test = test - 50;
            countChar(temp: test)
        }
        if (test >= 60 && test < 70){
            count = count + numbers.sixty.characters.count
            test = test - 60;
            countChar(temp: test)
        }
        if (test >= 70 && test < 80){
            count = count + numbers.seventy.characters.count
            test = test - 70;
            countChar(temp: test)
        }
        if (test >= 80 && test < 90){
            count = count + numbers.eighty.characters.count
            test = test - 80;
            countChar(temp: test)
        }
        if (test >= 90 && test < 100){
            count = count + numbers.ninety.characters.count
            test = test - 90;
            countChar(temp: test)
        }
        if (test >= 100 && test < 200){
            var hundred : String
            
            if( (test % 100) == 0){
                hundred = "\(numbers.one)\(numbers.hundred)"
            }else{
                hundred = "\(numbers.one)\(numbers.hundred)and"
            }
            count = count + hundred.characters.count
            test = test - 100;
            countChar(temp: test)
        }
        if (test >= 200 && test < 300){
            var hundred : String
            
            if( (test % 100) == 0){
                hundred = "\(numbers.two)\(numbers.hundred)"
            }else{
                hundred = "\(numbers.two)\(numbers.hundred)and"
            }
            count = count + hundred.characters.count
            test = test - 200;
            countChar(temp: test)
        }
        if (test >= 300 && test < 400){
            var hundred : String
            
            if( (test % 100) == 0){
                hundred = "\(numbers.three)\(numbers.hundred)"
            }else{
                hundred = "\(numbers.three)\(numbers.hundred)and"
            }
            count = count + hundred.characters.count
            test = test - 300;
            countChar(temp: test)
        }
        if (test >= 400 && test < 500){
            var hundred : String
            
            if( (test % 100) == 0){
                hundred = "\(numbers.four)\(numbers.hundred)"
            }else{
                hundred = "\(numbers.four)\(numbers.hundred)and"
            }
            count = count + hundred.characters.count
            test = test - 400;
            countChar(temp: test)
        }
        if (test >= 500 && test < 600){
            var hundred : String
            
            if( (test % 100) == 0){
                hundred = "\(numbers.five)\(numbers.hundred)"
            }else{
                hundred = "\(numbers.five)\(numbers.hundred)and"
            }
            count = count + hundred.characters.count
            test = test - 500;
            countChar(temp: test)
        }
        if (test >= 600 && test < 700){
            var hundred : String
            
            if( (test % 100) == 0){
                hundred = "\(numbers.six)\(numbers.hundred)"
            }else{
                hundred = "\(numbers.six)\(numbers.hundred)and"
            }
            count = count + hundred.characters.count
            test = test - 600;
            countChar(temp: test)
        }
        if (test >= 700 && test < 800){
            var hundred : String
            
            if( (test % 100) == 0){
                hundred = "\(numbers.seven)\(numbers.hundred)"
            }else{
                hundred = "\(numbers.seven)\(numbers.hundred)and"
            }
            count = count + hundred.characters.count
            test = test - 700;
            countChar(temp: test)
        }
        if (test >= 800 && test < 900){
            var hundred : String
            
            if( (test % 100) == 0){
                hundred = "\(numbers.eight)\(numbers.hundred)"
            }else{
                hundred = "\(numbers.eight)\(numbers.hundred)and"
            }
            count = count + hundred.characters.count
            test = test - 800;
            countChar(temp: test)
        }
        if (test >= 900 && test < 1000){
            var hundred : String
            
            if( (test % 100) == 0){
                hundred = "\(numbers.nine)\(numbers.hundred)"
            }else{
                hundred = "\(numbers.nine)\(numbers.hundred)and"
            }
            count = count + hundred.characters.count
            test = test - 900;
            countChar(temp: test)
        }
        if (test == 1000){
            var thousand : String = "\(numbers.one)\(numbers.thousand)"
            count += thousand.characters.count
        }
}

for i in stride(from: 0, to: 1001, by: 1){
//    countChar(temp: i)
    if(i < 19){
        count += onetoNineteen[i].characters.count
    }else{
        charCount(charCount: i)
    }
}

print(count)
// 21,124